pandas/core/sample.py

```diff
@@ -150,6 +150,11 @@ def sample(
         else:
             raise ValueError("Invalid weights: weights sum to zero")
 
+        if weight_sum > 1 and replace:
+            raise ValueError(
+                "Invalid weights: If `replace`=True weights must sum to less than 1"
+            )
```

Comment:
This is not the correct condition. See the formula in the OP of the linked issue.